api/src/test/java/com/codeforcommunity/rest/RestFunctionsTest.java

```
@@ -0,0 +1,171 @@
+package com.codeforcommunity.rest;
+
+import org.mockito.Mockito;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.core.http.HttpServerRequest;
+
+import com.codeforcommunity.dto.auth.*;
+import com.codeforcommunity.exceptions.*;
+
+// Contains tests for RestFunctions.java in main
+public class RestFunctionsTest {
+    HttpServerRequest mockRequest = Mockito.mock(HttpServerRequest.class);
+    RoutingContext mockRoutingContext = Mockito.mock(RoutingContext.class);
+
+    // Test for empty JSON response
+    @Test
+    public void testGetJsonBodyAsClass1() {
+        String emptyJSONString = "{}";
+        JsonObject emptyJSONObject = new JsonObject(emptyJSONString);
+
+        when(mockRoutingContext.getBodyAsJson())
+        .thenReturn(emptyJSONObject);
+
+        LoginRequest result = RestFunctions.getJsonBodyAsClass(mockRoutingContext, LoginRequest.class);
+
+        assertEquals(result.getEmail(), null);
```

Does this test pass? Because really it shouldn't. If the fields we're expecting aren't present then this should throw an exception. If it does pass then you found a bug already!

---

Yeah, it passes